src/screens/FeedScreen/RenderList.js

```diff
@@ -110,6 +119,87 @@ const RenderListFeed = (props) => {
       getCommentLength(item.latest_reactions.comment) > 0 ? getHeightReaction() / 2.2 : 0;
     return dimen.size.FEED_CURRENT_ITEM_HEIGHT - getHeightHeader() - getHeightFooter() - haveLength;
   };
+
+  const handleFollowUnfollow = async (value, isComment = false) => {
+    // is value true user aleady follow
+    // is value false user not follow yet
+
+    const user_id = isComment ? item?.latest_reactions?.comment[0].user_id : item?.actor?.id;
+    const username = isComment
+      ? item?.latest_reactions?.comment[0]?.user?.data?.username
+      : item?.actor?.data?.username;
+
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: user_id,
+      username_follower: myProfile?.username,
+      username_followed: username,
+      follow_source: 'feed'
+    };
+
+    const dataFollowAnon = {
+      follow_source: 'post',
+      post_id: item?.id
+    };
+
+    const indexFeed = feedsContext?.feeds?.findIndex((feed) => {
+      return feed?.id === item?.id;
+    });
+    const feedData = feedsContext?.feeds[indexFeed];
+
+    if (feedData?.is_following_target) {
+      if (!feedData?.anon_user_info_color_name) {
+        feedsContext?.feeds.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            setFeedByIndex(
+              {
+                index,
+                singleFeed: {...feedsContext?.feeds[index], is_following_target: false}
+              },
+              feedsContextDispatch
+            );
+          }
+        });
+        await setUnFollow(data);
+      } else {
+        const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: false};
+        setFeedByIndex(
+          {
+            index: indexFeed,
+            singleFeed: newFeed
+          },
+          feedsContextDispatch
+        );
+        await unfollowUserAnon(dataFollowAnon);
+      }
+    } else {
+      if (!feedData?.anon_user_info_color_name) {
+        feedsContext?.feeds.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            setFeedByIndex(
+              {
+                index,
+                singleFeed: {...feedsContext?.feeds[index], is_following_target: true}
+              },
+              feedsContextDispatch
+            );
+          }
+        });
+        await setFollow(data);
+      } else {
+        const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: true};
+        setFeedByIndex(
+          {
+            index: indexFeed,
+            singleFeed: newFeed
+          },
+          feedsContextDispatch
+        );
+        await followUserAnon(dataFollowAnon);
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new function `handleFollowUnfollow` introduces a lot of complexity. It's important to ensure that it handles all edge cases correctly and doesn't introduce any potential bugs or performance issues. Also, consider breaking down this function into smaller, more manageable functions to improve readability and maintainability.
```diff
+ const handleFollowUnfollow = async (value, isComment = false) => {
+   // ... function implementation ...
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->